ge/org.osate.ge/src/org/osate/ge/DiagramExporter.java

```diff
@@ -0,0 +1,296 @@
+		return new GraphitiAgeDiagram(ageDiagram, diagram, editingDomain, c -> {
+			editingDomain.getCommandStack().execute(new AbstractCommand() {
+				@Override
+				protected boolean prepare() {
+					return true;
+				}
+
+				@Override
+				public void execute() {
+					c.execute();
+				}
+
+				@Override
+				public boolean canUndo() {
+					return false;
+				}
+
+				@Override
+				public void redo() {
+				}
+			});
+
+		}, coloringProvider, () -> {
+		});
+	}
+
+	// Shift elements
+	private static void moveDiagramElements(final ContainerBounds containerBounds, final Diagram diagram) {
+		final int xOffset = -containerBounds.minX + containerBounds.padding;
+		final int yOffset = -containerBounds.minY + containerBounds.padding;
+		for (final Shape s : diagram.getChildren()) {
+			final GraphicsAlgorithm ga = s.getGraphicsAlgorithm();
+			ga.setX(xOffset + ga.getX());
+			ga.setY(yOffset + ga.getY());
+		}
+
+		for (final Connection c : diagram.getConnections()) {
+			if (c instanceof FreeFormConnection) {
+				final FreeFormConnection ffc = (FreeFormConnection) c;
+				for (final Point p : ffc.getBendpoints()) {
+					p.setX(xOffset + p.getX());
+					p.setY(yOffset + p.getY());
+				}
+			}
+		}
+	}
+
+	private static void setDiagramContainerBounds(final ContainerBounds containerBounds, final Diagram diagram) {
+		for (final Shape shape : diagram.getChildren()) {
+			final GraphicsAlgorithm ga = shape.getGraphicsAlgorithm();
+			compareBounds(containerBounds, ga.getX(), ga.getX() + ga.getWidth(), ga.getY(), ga.getY() + ga.getHeight());
+		}
+
+		compareConnectionBounds(containerBounds, diagram.getConnections());
+	}
+
+	// Compare points of connections to ensure that connections are inside image bounds
+	private static void compareConnectionBounds(final ContainerBounds containerBounds,
+			final EList<Connection> connections) {
+		for (final Connection c : connections) {
+			if (c instanceof FreeFormConnection) {
+				final FreeFormConnection fc = (FreeFormConnection) c;
+				for (final Point p : fc.getBendpoints()) {
+					compareBounds(containerBounds, p.getX(), p.getY());
+				}
+			}
+
+			compareConnectionDecoratorBounds(containerBounds, c);
+		}
+	}
+
+	// Make sure connection decorators are contained within the image
+	private static void compareConnectionDecoratorBounds(final ContainerBounds containerBounds, final Connection c) {
+		for (final ConnectionDecorator cd : c.getConnectionDecorators()) {
+			if (cd.isLocationRelative()) {
+				final GraphicsAlgorithm labelGA = cd.getGraphicsAlgorithm();
+				final ILocation loc = GraphitiUi.getPeService().getConnectionMidpoint(c, cd.getLocation());
+				final int labelX = loc.getX() + labelGA.getX();
+				final int labelY = loc.getY() + labelGA.getY();
+				compareBounds(containerBounds, labelX, labelX + labelGA.getWidth(), labelY, labelY + labelGA.getY());
+			}
+		}
+	}
+
+	private static void compareBounds(final ContainerBounds containerBounds, final int x, final int y) {
```

Comment:
Neither this method or the other compareBounds() method side effect is to compare bounds. The method extends the bounds to include the specified range. The method should be renamed and comments should be updated accordingly. Suggested names: extendBounds() or updateBounds().